nrf-hal-common/src/pwm.rs

```diff
@@ -473,37 +474,29 @@ where
 
     /// Loads a sequence buffer.
     /// NOTE: `buf` must live until the sequence is done playing, or it might play a corrupted sequence.
-    pub fn load_seq(&self, seq: Seq, buf: &[u16]) -> Result<(), Error> {
-        if (buf.as_ptr() as usize) < SRAM_LOWER || (buf.as_ptr() as usize) > SRAM_UPPER {
+    pub fn load_seq<B>(&self, seq: Seq, buf: B) -> Result<(), Error>
+    where
+        B: ReadBuffer<Word = u16>,
+    {
+        let (ptr, len) = unsafe { buf.read_buffer() };
```

Comment:
The safety contract here seems to be the caller's responsibility, so this `unsafe` doesn't look correct here (the caller could call `&mut self` methods on the buffer if it passes `&[u16]`).

Well, I suppose it's not any worse than before though :)

Reply:
Hmm, sorry i don't quite follow what you suggest changing?

Reply:
Something like the `Transfer` api that we use in the other DMA implementations? The problem i saw in this instance was how to make it sensible to use, with all the possible sequence playback variants / loops etc...

Reply:
Yeah, my issue is that the `'static` bound on the buffer is missing, basically. I know that technically this is fine (at worst PWM will read garbage data), but that means that using the DMA traits is not really necessary here either.